src/context/AnimationsContext.js

```diff
@@ -0,0 +1,30 @@
+import PropTypes from 'prop-types';
+import React, { createContext, useContext, useState } from 'react';
+
+const AnimationsContext = createContext();
+
+export const useAnimation = () => useContext(AnimationsContext);
```

Comment:
Bardzo ciekawy pomysł na wyeksportowanie contextu, nie widziałem takiego nigdy!

Tylko jedna rzecz, proponuję zmianę nazwy na useAnimationContext, tak aby było wiadomo, że to context.